fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java

```diff
@@ -92,9 +105,20 @@ public void checkAndSetSupportPartitionInformation() {
         }
     }
 
+    private HikariDataSource createHikariDataSource() {
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(properties.get(JDBCResource.URI));
+        config.setUsername(properties.get(JDBCResource.USER));
+        config.setPassword(properties.get(JDBCResource.PASSWORD));
+        config.setDriverClassName(properties.get(JDBCResource.DRIVER_CLASS));
+        config.setMaximumPoolSize(Config.jdbc_connection_pool_size);
+        config.setMinimumIdle(Config.jdbc_minimum_idle_connections);
+        config.setIdleTimeout(Config.jdbc_connection_idle_timeout_ms);
+        return new HikariDataSource(config);
+    }
+
     public Connection getConnection() throws SQLException {
-        return DriverManager.getConnection(properties.get(JDBCResource.URI),
-                properties.get(JDBCResource.USER), properties.get(JDBCResource.PASSWORD));
+        return dataSource.getConnection();
```

Comment:
Do we need to add a session variable to control whether to use the connection pool?

Reply:
I don't see any reason to use session variable.  connection pool is always a better option. Don't think you so?

Reply:
I have found that there may be an issue with connection related components.  may we provide a way to quickly recover?